Description: sql migration: ensure using innodb utf8 for assignment table
 During migration from 37 to 38, migration fails to create assignment table if
 we don't explictly specify InnoDB and UTF8 charset.
Author: Emilien Macchi <emilien.macchi@enovance.com>
Date: Mon, 19 May 2014 12:40:44 +0000 (+0200)
X-Git-Url: https://review.openstack.org/gitweb?p=openstack%2Fkeystone.git;a=commitdiff_plain;h=18289b9e26ba900302874e72fcbf4529580a973a
Bug-Ubuntu: https://launchpad.net/bugs/1320855
Signed-off-by: Emilien Macchi <emilien.macchi@enovance.com>
Change-Id: Icd88202d2f020aea02514e49c2d07e349c74949f
Origin: https://review.openstack.org/#/c/94187/

diff --git a/keystone/common/sql/migrate_repo/versions/038_add_assignment_table.py b/keystone/common/sql/migrate_repo/versions/038_add_assignment_table.py
index 92e9faa..b39fa6e 100644
--- a/keystone/common/sql/migrate_repo/versions/038_add_assignment_table.py
+++ b/keystone/common/sql/migrate_repo/versions/038_add_assignment_table.py
@@ -39,7 +39,9 @@ def upgrade(migrate_engine):
         sql.Column('role_id', sql.String(64), sql.ForeignKey('role.id'),
                    nullable=False),
         sql.Column('inherited', sql.Boolean, default=False, nullable=False),
-        sql.PrimaryKeyConstraint('type', 'actor_id', 'target_id', 'role_id'))
+        sql.PrimaryKeyConstraint('type', 'actor_id', 'target_id', 'role_id'),
+        mysql_engine='InnoDB',
+        mysql_charset='utf8')
     assignment_table.create(migrate_engine, checkfirst=True)
 
 
